sw/device/lib/dif/dif_uart.c

```diff
@@ -98,10 +98,10 @@ static void uart_reset(const dif_uart_t *uart) {
 /**
  * Performs fundamental UART configuration.
  */
-static bool uart_configure(const dif_uart_t *uart,
-                           const dif_uart_config_t *config) {
+static dif_uart_config_result_t uart_configure(
+    const dif_uart_t *uart, const dif_uart_config_t *config) {
   if (config->baudrate == 0 || config->clk_freq_hz == 0) {
-    return false;
+    return kDifUartConfigBadConfig;
```

Comment:
Shouldn't this be something like `kDifUartConfigBadArg`?

Reply:
Well, those are config values, so my thinking was if config data is wrong - it's "bad config".

Reply:
Ah, I now see this is reasonable, given `uart_configure` is not a public function.